sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Listeners/EventHubListener.PartitionProcessor.cs

```diff
@@ -218,13 +219,29 @@ public async Task ProcessEventsAsync(EventProcessorHostPartition context, IEnume
                 // so that is the responsibility of the user's function currently. E.g.
                 // the function should have try/catch handling around all event processing
                 // code, and capture/log/persist failed events, since they won't be retried.
-                if (eventToCheckpoint != null)
+                if (eventToCheckpoint == null)
                 {
-                    await CheckpointAsync(eventToCheckpoint, context).ConfigureAwait(false);
+                    return;
                 }
+
+                // If we are shutting down (or have lost ownership) and the function invocation failed, we should not checkpoint
+                // as there is a possibility the failure caused the user's function to not execute. We use the token rather than
+                // the _stoppedReason because the _stoppedReason is not set until after the partition processor has stopped.
```

Comment:
Moving this so it's not confusing with the other conversation - are there cases where the `partitioncancellationtoken` is canceled but we're not shutting down/ ownership isn't lost? Just curious whether this should be just the _cts token or the linked one.

Reply:
I can't think of any cases where that would be true but wanted to throw it out there.

Reply:
My understanding of the partitionCancellationToken is that it will be signaled when the processor is shutting down OR when partition ownership is lost. That said, I'm removing the stopped reason logic so we will just not checkpoint if the linked token is signaled (regardless of the reason).